---
title: "Differential Gene Expression Analysis and Preliminary ORA"
subtitle: "BCB420 Assignment 2"
author: Jielin Yang
date: March 14, 2023
output: 
  html_document:
    toc: true
    toc_depth: 3
    toc_float: true
    number_sections: true
    df_print: paged
    highlight: pygments
bibliography: Assignment2.bib
csl: "american-medical-association.csl"
---

<style type="text/css">
    body {
    max-width: 100%;
    overflow-x: auto;
    font-family: Helvetica;
    font-size: 12pt;
    }
    h1 {
    font-size: 20pt;
    font-weight: bold;
    }
    h2 {
        font-size: 18pt;
        font-weight: bold;
    }
    h3 {
        font-size: 16pt;
        font-weight: bold;
    }
    h4 {
        font-size: 14pt;
        font-weight: bold;
    }
    code {
        font-family: monospace;
        font-size: 11pt;
    }
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(root.dir = here::here())  # Must be the project directory
```

# Introduction

## Background

Heart failure (HF) is a chronic cardiac disease that has progressively severe implications
on the patient's quality of life, and it has been a leading cause of death and a significant
contributor to the global burden of disease. [@heidenreich20222022] The pathophysiology of HF 
is complex and involves multiple factors, but a major hallmark of HF is the progressive
degeneration of the myocardium that is accompanied by cardiac fibrosis. [@garoffolo2022reduction] 
[@porter2009cardiac] [@tallquist2017redefining] During the progression of HF, a process known
as cardiac remodelling progressively replaces the myocardium with fibrous tissues, which in turn
leads to the gradual stiffening of the heart ventricular wall. Thus, the identification of
profibrotic pathways are essential to the development of therapeutic strategies to delay or
even to prevent the profibrotic remodelling of the heart.

Previous studies have shown that the straining condition has played a significant role in
potently altering the Hippo pathways. [@brusatin2018biomaterials] It was found that mechanical
stress on the cell cytoskeleton induces the nuclear entry of the YAP/TAZ complex, which 
subsequently activates the proliferative pathways. [@elosegui2017force] [@panciera2017mechanobiology]
Although such a mechanism may seem to be beneficial for the restoration of the damaged cardiomyocytes,
this pathway has been shown to be more active in cardiac stromal cells, which can subsequently
differentiate into myofibroblasts that eventually leads to the profibrotic remodelling of the heart. 
[@garoffolo2022reduction] Therefore, directly suppressing the YAP/TAZ complex in cardiac stromal
cells may be an effective strategy. Recently, a study by [Garoffolo et al.](https://www.ahajournals.org/doi/10.1161/CIRCRESAHA.121.319373) 
investigated the use of a small molecular inhibitor, verteporfin (VTP), to suppress the YAP/TAZ complex 
with or without the presence of TGF-β1, a profibrotic cytokine under the maximal strain condition.

## Objectives

The study by [Garoffolo et al.](https://www.ahajournals.org/doi/10.1161/CIRCRESAHA.121.319373) aimed
to examine the antifibrotic effect of VTP in cardiac stromal cells under profibrotic signals that 
simulates the straining conditions under heart failure as part of their larger objective. [@garoffolo2022reduction]

Here, we focus on this particular aspect of the research discussed above. In particular, we aim to 
use RNA-seq data to identify the changes in the transcriptional profile of cardiac stromal cells
after the use of VTP. [Figure 1.](#fig1) shows the experimental design of the study. Here, we perform 
differential gene expression analysis comparing the transcriptome of the cells treated with VTP
with control under the known profibrotic straining condition. Additionally, we test the efficacy
of the drug (VTP) under the presence of TGF-β1, a potent profibrotic cytokine. [@saadat2021pivotal] To investigate whether
direct inhibition of the YAP/TAZ complex with VTP, we perform threshold over-representation analysis
to identify enriched pathways in the differentially expressed genes, which would allow us to identify
whether VTP can effectively reverse a profibrotic phenotype.

<a name="fig1"></a>
![](https://github.com/bcb420-2023/Jielin_Yang/raw/main/images/schematics.png)

**Figure 1.** Schematics of RNAseq experiment design. Samples were cultured under maximal mechanical strains. 
Six independent cell cultures were used for each treatment condition. cSt-Cs: cardiospheres-derived primitive 
cardiac stromal cells; VTP: verteporfin; TGF-β1: transforming growth factor beta-1. Figure adapted from 
[Assignment 1](https://github.com/bcb420-2023/Jielin_Yang/blob/main/Dataset%20selection%20and%20initial%20processing/Assignment_1_Data_cleaning_and_identifier_mapping.html) 
and created with BioRender.com.

## Dataset

The data used in the present analysis is retrieved from the GEO database under the accession number
[GSE203358](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE203358). The dataset contains
RNA-seq data from 24 samples, which are evenly split into 4 groups:
- Control (maximal strain condition)
- Verteporfin (VTP) treatment (maximal strain condition)
- TGF-β1 treatment (maximal strain condition)
- Verteporfin (VTP) treatment with TGF-β1 (maximal strain condition)

## Previous Analysis

Since we have previously performed initial data cleaning and normalization, here we provide a brief
summary of the previous analysis. The entire process can be found [here](https://github.com/bcb420-2023/Jielin_Yang/blob/main/Dataset%20selection%20and%20initial%20processing/Assignment_1_Data_cleaning_and_identifier_mapping.html).

### Data QC and Filtering

The original dataset is composed of `24` samples and `60583 `genes. After removing genes with low counts, the dataset
we use in the present analysis is composed of `14814` genes. The reference genome used is the human [GRCh38](https://useast.ensembl.org/Homo_sapiens/Info/Index?db=core) genome.

### Identifier Mapping and Normalization

The original dataset contains a mix of HUGO gene symbols and GenBank identifiers. We used the [biomaRt](https://bioconductor.org/packages/release/bioc/html/biomaRt.html) 
package to map the rest of the identifiers to HUGO gene symbols. [@durinck2009mapping] The dataset is normalized using the Trimmed
Mean of M-values (TMM) method implemented by the [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html) package. [@robinson2010edger]
After applying normalization, the library size for each sample has been effectively adjusted such
that the distribution of gene expressions are comparable across samples.

### Multidimensional Scaling

We evaluated the similarity between the samples by accessing the clustering of the samples (as well 
as replicates within each group) using multidimensional scaling (MDS). The MDS plot shown in [Figure 2.](#fig2) reveals a moderate
clustering of samples based on the treatment conditions, and in particular, the control samples
under mechanical strain alone are clustered separately with any other groups. The other treatment
groups show less separation along the first dimension, but clustering is still observed. Interestingly,
although most of the samples are clustered based on the treatment conditions and that we did not
observe any clustering based on the cell culture replicates, we notice that certain samples that
do not cluster tightly with the rest in the same treatment group seem to be consistent across the 
conditions. For example, comparing to other replicates, in cells from culture `M89` and `M99`,
different conditions seem to have a smaller effect on the gene expression profile.

<a name="fig2"></a>
![](https://github.com/bcb420-2023/Jielin_Yang/raw/main/images/mds.png)

**Figure 2.** Multidimensional scaling plot of cSt-Cs gene expression across samples. All samples
are coloured by treatment condition. The first two dimensions are shown. Moderate clustering
by treatment and low clustering by replicate for a small number of cultures is observed. Figure
adapted from [Assignment 1](https://github.com/bcb420-2023/Jielin_Yang/blob/main/Dataset%20selection%20and%20initial%20processing/Assignment_1_Data_cleaning_and_identifier_mapping.html).

# Preparations

## Package Dependencies

The current R Notebook assumes that it is being run on the BCB420-2023 base image. 
Additional packages required are noted below.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# edgeR package for manupulating and analyzing RNAseq data
if (! requireNamespace("edgeR", quietly = TRUE)) {
  BiocManager::install("edgeR")
}

# ComplexHeatmap package for creating heatmaps
if (! requireNamespace("ComplexHeatmap", quietly = TRUE)) {
  BiocManager::install("ComplexHeatmap")
}

# circlize package for circular visualization
if (! requireNamespace("circlize", quietly = TRUE)) {
  BiocManager::install("circlize")
}

# KableExtra package for creating tables and formatting
if (! requireNamespace("kableExtra", quietly = TRUE)) {
  install.packages("kableExtra")
}

# dplyr package for data manipulation
if (! requireNamespace("dplyr", quietly = TRUE)) {
  install.packages("dplyr")
}

# DT package for creating interactive tables
if (! requireNamespace("DT", quietly = TRUE)) {
  install.packages("DT")
}

# Load the dplyr package
library("dplyr")

# We will consistently use package::function() to avoid confusion
# However, the dplyr package has to be loaded first to allow
# definitions of various operators such as %>%
```

## Data Import

In the previous analysis, we have saved the processed data output to a file. 
Hence, we can simply load the data from the file and perform the analysis.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Load the processed data
counts <- read.csv(file = "./Data/expr_norm_counts.csv", header = TRUE, row.names = 1)

# View the first few rows of the data
head(counts)
```

**Table 1.** Processed count data used for the present analysis. The data is composed of
14814 genes and 24 samples, where each gene is identified by a unique HUGO symbol.
The first few rows are shown.

Since the header of the data contains the sample information, we will extract the sample
information from the header to reconstruct the experimental design.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Extract the sample information from the header
sample_matrix <- colnames(counts) %>%
  lapply(function(x){unlist(strsplit(x, split = "\\."))}) %>%
  as.data.frame()

# Rename the data frame
colnames(sample_matrix) <- colnames(counts)
rownames(sample_matrix) <- c("culture","treatment")

# Transpose the data frame so that each row represents a unique sample
# and each column represents a unique sample attribute
sample_matrix <- data.frame(t(sample_matrix))

# Order the samples by their treatment conditions
sample_matrix <- sample_matrix[order(sample_matrix$treatment), ]

# View the sample information
DT::datatable(sample_matrix, rownames = TRUE, filter = "top",
              options = list(pageLength = 6)) %>%
              # Changing the font size of content
              DT::formatStyle(names(sample_matrix), fontSize = "12px") %>%
              # Changing the font size of row names (sample names)
              DT::formatStyle(0, fontSize = "12px")
```

**Table 2.** Sample information used for the present analysis. A total of 24 samples
are included in the analysis, where each sample is identified by a unique culture ID
and treatment condition. Cells are retrieved from 6 independent cultures, and each
is used for 4 different treatment conditions.

# Differential Gene Expression Analysis

Understanding the antifibrotic effect of VTP requires a comprehensive analysis of the
gene expression profile of the cSt-Cs by identifying the transcriptional changes with
or without the treatment of VTP. To this end, we will perform two separate differential
gene expression analysis with or without the presence of TGF-β1, in which we can isolate
the effect of VTP as the only independent variable.

However, although our data is composed of a total of 4 treatment condition, which allows
for the comparison between control (strain alone) and TGF-β1, we will not perform such
analysis. It has been well documented that TGF-β1 is a potent profibrotic cytokine that
will induce a series of transcriptional activation on profibrotic pathways. [@walker2019transcriptomic]
The authors of the original publication has also verified the profibrotic effect of TGF-β1. [@saadat2021pivotal]
We have also shown that TGF-β1 significantly upregaltes the expression of profibrotic
genes (see associated [journal entry](https://github.com/bcb420-2023/Jielin_Yang/wiki/Differential-Gene-Expression-Analysis-and-Preliminary-ORA#differential-gene-expression-analysis-comparing-tgf%CE%B21-and-control) 
for details). Therefore, based on our objective, we believe that it is more valuable to 
investigate the effect of VTP under maximal strain, both with and without the presence of
TGF-β1 separately.

## Global Gene Expression Profiling

Before we perform the differential gene expression analysis, we first want to
observe the global expression change at the gene level. Here, we plot a heatmap
for all the genes in the dataset to visualize the expression change across all
samples under various conditions.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Creating a DGEList object
counts_dge <- edgeR::DGEList(counts = counts, group = sample_matrix$treatment)

# As discussed in the associated journal entry, the data we use are the
# counts, rather than the counts per million (CPM) values. Therefore, we
# will need to convert the count to CPM values for normalization on
# library size.
counts_dge <- edgeR::calcNormFactors(counts_dge)
counts_cpm_all <- edgeR::cpm(counts_dge)
```

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Create a heatmap matrix for all the genes by scaling the CPM values
# This allows the heatmap colour to be on a consistent and continuous scale
# across all the samples
heatmap_matrix <- counts_cpm_all %>% t() %>% scale()

# Convert it back to the original format
heatmap_matrix <- t(heatmap_matrix)
```

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Checking the range of the heatmap matrix
hmap_range <- range(heatmap_matrix)
hmap_range
```

The range of the heatmap matrix is between `r hmap_range[1]` and `r hmap_range[2]`.
Since the scaled expression value ranges from negative to positive, we will define
a three colour palette for the heatmap, where we will use `blue` to represent
the downregulated genes and `red` for upregulated genes. In the middle of the scale
is `white`.

Here, heatmap creation and visual representation are implemented using the
`ComplexHeatmap` and `circlize` packages, respectively. [@gu2016complex] [@gu2014circlize]

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Set the 3 color palette based on the scaled heatmap matrix
hmap_col <- circlize::colorRamp2(c(hmap_range[1], 0, hmap_range[2]), 
                                 c("blue", "white", "red"))
```

Due to the relatively large number of samples, we will also apply annotations to
the heatmap to better visualize the clustering of the samples. In particular,
we will annotate the heatmap by the treatment condition and culture replicate.
This would better allows us to identify whether any of the variables have a
significant effect on the clustering of the samples.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Treatment Conditions
treatments <- unique(sample_matrix$treatment)

# Define 4 colours for the 4 treatment conditions
treatment_col <- hcl.colors(4, palette = "viridis")
names(treatment_col) <- treatments

# Culture Replicates
cultures <- unique(sample_matrix$culture)

# Define 6 colours for the 6 culture replicates
culture_col <- hcl.colors(6, palette = "harmonic")

# Create an annotation dataframe
anno_df <- data.frame(treatment = sample_matrix$treatment,
                      culture = sample_matrix$culture)
anno_col <- list(treatment = treatment_col, culture = culture_col)

# Create the heatmap annotation
hmap_anno <- ComplexHeatmap::HeatmapAnnotation(
                    df = anno_df,
                    col = anno_col)

```






To better visualize the heatmap, we cluster the genes into 4 groups using the
k-means clustering algorithm. This allows the genes that have similar expression
profiles to be grouped together. Additionally, since our dataset consists of 4
treatment conditions, it is expected that a clustering of 4 groups will reveal
some interesting patterns in the expression profiles.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Generate the heatmap
# Set the seed for reproducibility due to he inherent randomness of the
# k-means clustering algorithm
set.seed(91711)
hmap_all <- ComplexHeatmap::Heatmap(heatmap_matrix,
                                    show_row_dend = TRUE,
                                    show_column_dend = TRUE,
                                    show_column_names = TRUE,
                                    show_row_names = FALSE,
                                    col = hmap_col,
                                    km = 4)
set.seed(NULL)

# Plot the heatmap
hmap_all
```
## Dispersion Estimation

## Differential expression

### Effect of VTP under Maximal Strain

### Effect of VTP under Maximal Strain with Profibrotic TGF-β1




# Threshold Over-representation Analysis

## Effect of VTP under Maximal Strain

### All Differentially Expressed Genes

### Up-regulated Genes in VTP

### Down-regulated Genes in VTP


## Effect of VTP under Maximal Strain with Profibrotic TGF-β1

### All Differentially Expressed Genes

### Up-regulated Genes in VTP

### Down-regulated Genes in VTP




# Interpretation and Analysis

## Differential Gene Expression Analysis

- **How many genes were significantly differentially expressed? What thresholds did you use and why?**





- **Multiple hypothesis testing - correct your p-values using a multiple hypothesis correction method. Which method did you use? And Why? How many genes passed correction?**

- **Show the amount of differentially expressed genes using an MA Plot or a Volcano plot. Highlight genes of interest.**

- **Visualize your top hits using a heatmap. Do you conditions cluster together? Explain why or why not.**


## Threshold Over-representation Analysis

- **Which method did you choose and why?**

- **What annotation data did you use and why? What version of the annotation are you using?**

- **How many genesets were returned with what thresholds?**

- **Run the analysis using the up-regulated set of genes, and the down-regulated set of genes separately. How do these results compare to using the whole list (i.e all differentially expressed genes together vs. the up-regulated and down regulated differentially expressed genes separately)?**


## Result Interpretation

- **Do the over-representation results support conclusions or mechanism discussed in the original paper?**

- **Can you find evidence, i.e. publications, to support some of the results that you see. How does this evidence support your results.**

# Journal Link
The link to the concurrent journal entry can be found [here](https://github.com/bcb420-2023/Jielin_Yang/wiki/Differential-Gene-Expression-Analysis-and-Preliminary-ORA).

# References