---
title: "Gene Set Enrichement and Network Analysis"
subtitle: "BCB420 Assignment 3"
author: Jielin Yang
date: April 4th, 2023
output: 
  html_document:
    toc: true
    toc_depth: 3
    toc_float: true
    number_sections: true
    df_print: paged
    highlight: pygments
bibliography: Assignment3.bib
csl: "american-medical-association.csl"
---

<style type="text/css">
    body {
    max-width: 100%;
    overflow-x: auto;
    font-family: Helvetica;
    font-size: 12pt;
    }
    h1 {
    font-size: 20pt;
    font-weight: bold;
    }
    h2 {
        font-size: 18pt;
        font-weight: bold;
    }
    h3 {
        font-size: 16pt;
        font-weight: bold;
    }
    h4 {
        font-size: 14pt;
        font-weight: bold;
    }
    code {
        font-family: monospace;
        font-size: 11pt;
    }
</style>

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(root.dir = here::here())  # Must be the project directory
```

```{r message=FALSE, warning=FALSE, include=FALSE}
# Run A2 to generate the required data
rmarkdown::render("./A2_Jielin_Yang.Rmd")
```

# Introduction

## Background

Heart failure (HF) is a chronic cardiac disease with complex pathophysiology 
with profound implications in the patient's quality of life. The progressively
degenerating myocardium is a major hallmark of HF, which is accompanied by
cardiac fibrosis. [@garoffolo2022reduction] [@porter2009cardiac] [@tallquist2017redefining]
Therefore, the identification of profibrotic pathways is essential to the
understanding the pathophysiology of HF and the development of therapeutic
strategies to delay or even to prevent the profibrotic remodelling of the heart.

The Hippo pathway is a conserved signalling pathway that regulates cell
proliferation, cell survival, and cell migration. [@mia2019hippo] It has been
previously discovered that a central component of this pathway, the YAP/TAZ
complex, is dysregulated upon mechanimal stimulation of the cytoskeleton. 
[@elosegui2017force] [@panciera2017mechanobiology] [@brusatin2018biomaterials]
Interestingly, heart failure is usually accompanied by mechanical stress due to 
the malfunctioning of the pumping capability of the heart. Therefore, the
direct suppression of the YAP/TAZ complex may be a potential therapeutic
approach to suppress the proliferation of pro-fibrotic cells and to prevent
the profibrotic remodelling of the heart. [@garoffolo2022reduction] Recently, 
[Garoffolo et al.](https://www.ahajournals.org/doi/10.1161/CIRCRESAHA.121.319373) 
investigated the use of a small molecular inhibitor, verteporfin (VTP) in
cardiac stromal cells, an important precursor to cardiac fibroblasts, to 
suppress the YAP/TAZ complex with or without the presence of TGF-β1, a 
profibrotic cytokine under the maximal strain condition.

## Objectives

In the present analysis, we aim to use RNA-seq data to identify the changes in
the transcriptional profile of cardiac stromal cells after the use of VTP. As
shown in [Figure 1.](#fig1), the transcriptome was assessed by comparing the
addition of VTP with the control with or without the presence of the profibrotic
cytokine, TGF-β1. [@saadat2021pivotal] Upon the identification of differentially
expressed genes, we aim to perform gene set enrichment analysis (GSEA) and
network analysis to access the enriched pathways and using it to infer the
molecular mechanisms underlying the effects of VTP on cardiac stromal cells.

<a name="fig1"></a>
![](https://github.com/bcb420-2023/Jielin_Yang/raw/main/images/schematics.png)

**Figure 1.** Schematics of RNAseq experiment design. Samples were cultured under 
maximal mechanical strains. Six independent cell cultures were used for each 
treatment condition. cSt-Cs: cardiospheres-derived primitive cardiac stromal 
cells; VTP: verteporfin; TGF-β1: transforming growth factor beta-1. Figure 
adapted from [Assignment 1](https://github.com/bcb420-2023/Jielin_Yang/blob/main/Dataset%20selection%20and%20initial%20processing/Assignment_1_Data_cleaning_and_identifier_mapping.html) 
and created with BioRender.com.

## Dataset

The data used in the present analysis is retrieved from the GEO database under 
the accession number [GSE203358](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE203358). 
The dataset contains RNA-seq data from 24 samples, which are evenly split into 4 
groups:
- Control (maximal strain condition)
- Verteporfin (VTP) treatment (maximal strain condition)
- TGF-β1 treatment (maximal strain condition)
- Verteporfin (VTP) treatment with TGF-β1 (maximal strain condition)

## Previous Analysis

### Data Normalization

Upon filtering out genes with low counts, the dataset is 
composed of `14814` genes and `24` samples. The reference genome used is the 
human [GRCh38](https://useast.ensembl.org/Homo_sapiens/Info/Index?db=core).

The dataset was normalized using the Trimmed Mean of M-values (TMM) method 
implemented by the [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html) 
package. [@robinson2010edger] After applying normalization, the library size for 
each sample has been effectively adjusted such that the distribution of gene 
expressions are comparable across samples.

### Cluster Analysis

Samples were clustered based on transcriptional similarity. multidimensional
scaling analysis indicates distinct separation based on treatment conditions,
with the most highly distinguishable cluster of samples in the control group,
without profibrotic TGF-β1 treatment. Additional hierarchical clustering with
heatmap further supports that the control group has the most distinguishable
transcriptome, with few samples having similar expression patterns with the 
other groups. VTP treatment are found to alter the transcriptional profile 
to an high extent, but its effect is less prominent when combined with TGF-β1.
In addition, the samples show moderate to low degree of clustering based on the 
cell culture replicates, suggesting that the variation between samples are less
driven by the intrinsic biological variation between the cell cultures.

### Differential Expression Analysis and Preliminary ORA

Differential gene expression analysis was performed using the `edgeR` package. 
[@robinson2010edger] [Figure 2.](#fig2) shows the volcano plot of the differentially
expressed genes (DEGs) between the control and VTP treatment groups, with the
top enriched GO: Biological Process terms in the significantly downregulated 
genes. In this analysis, `r nrow(deg_vtp_ctrl)` genes are found to be differentially
expressed between the control and VTP treatment group, without the presence of
TGF-β1. Among these genes, `r nrow(deg_vtp_ctrl[deg_vtp_ctrl$logFC > 0,])` are
significantly upregulated and `r nrow(deg_vtp_ctrl[deg_vtp_ctrl$logFC < 0,])` are
significantly downregulated. Many of the downregulated genes are collagen genes,
which are a major component of the extracellular matrix (ECM) and are involved
in the profibrotic remodelling of the heart. [@garoffolo2022reduction] The top
enriched GO: Biological Process terms are also consistent with such findings, 
showing pathways in cell adhesion and ECM organization. However, when TGF-β1 is
added, the effect of VTP is much less pronounced, with only `r nrow(deg_vtp_tgf)`
genes being differentially expressed. 

<a name="fig2"></a>

```{r fig.height=4, fig.width=10, echo=FALSE, fig.align="center"}
# Replot the dot plot from assignment 2 with only the top 5 enriched GO terms
# in the significantly downregulated genes
gobp_downreg_5 <- gs_downreg_10[1:5,]
dot_plot <- ggplot2::ggplot(gobp_downreg_5) + 
                ggplot2::geom_point(ggplot2::aes(x = gene_ratio, 
                            y = reorder(stringr::str_wrap(Term, 20), gene_ratio),
                            color = FDR,
                            size = Fold.Enrichment)) +
                ggplot2::theme_bw() +
                ggplot2::theme(axis.text.x = ggplot2::element_text(size = 10),
                               axis.text.y = ggplot2::element_text(size = 10),
                               axis.title.x = ggplot2::element_text(size = 10),
                               axis.title.y = ggplot2::element_text(size = 10)) +
                ggplot2::scale_color_gradient(low = "red", high = "blue") +
                ggplot2::labs(x = "Gene Ratio", y = NULL,
                              color = "FDR", size = "Fold Enrichment")

# Replot the volcano plot from assignment 2
vol_plot <- ggplot2::ggplot(results_vtp_ctrl, aes(logFC, -log(FDR, 10))) +
    ggplot2::geom_point(aes(color = Expression), size = 2/5) +
    xlab(expression("log"[2]*"FC")) + 
    ylab(expression("-log"[10]*"FDR")) + 
    ggplot2::scale_color_manual(values = c("Up-regulated" = "firebrick3", 
                                           "Down-regulated" = "dodgerblue3", 
                                           "Not DE" = "gray50")) +
    ggplot2::guides(color = ggplot2::guide_legend(override.aes = list(size=1.5))) +
    ggrepel::geom_label_repel(data = genes_to_label, 
                        mapping = aes(logFC, -log(FDR, 10), label = rownames(genes_to_label)),
                        size = 2.5, color = "black", nudge_x = 0.1, nudge_y = 0.1)

# Draw the plots
cowplot::plot_grid(vol_plot, dot_plot, ncol = 2, labels = "AUTO")
```

**Figure 2.** Differentially expressed genes and enriched GO: Biological Process terms
in the significantly downregulated genes. (A) Volcano plot of the differentially
expressed genes between the control and VTP treatment groups in the absence of
TGF-β1. Important genes are labelled, with most collagen genes found to be 
downregulated. (B) Dot plot of the top 5 enriched GO: Biological Process terms
in the significantly downregulated genes.


# Preparations

## Package Dependencies

The current R Notebook assumes that it is being run on the BCB420-2023 base image. 
Additional packages required are noted below. For reproducibility we list all packages
we used in the analysis, including utility packages. The GSEA and Cytoscape softwares
with the EnrichmentMap pipeline are also required for the analysis, but they are
run manually outside of the R Notebook.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# ggplot2 package for creating plots
if (! requireNamespace("ggplot2", quietly = TRUE)) {
  install.packages("ggplot2")
}
# cowplot package for arranging plots
if (! requireNamespace("cowplot", quietly = TRUE)) {
  install.packages("cowplot")
}
# ggrepel package for plotting text with minimal overlap
if (! requireNamespace("ggrepel", quietly = TRUE)) {
  install.packages("ggrepel")
}
# KableExtra package for creating tables and formatting
if (! requireNamespace("kableExtra", quietly = TRUE)) {
  install.packages("kableExtra")
}
# dplyr package for data manipulation
if (! requireNamespace("dplyr", quietly = TRUE)) {
  install.packages("dplyr")
}
# DT package for creating interactive tables
if (! requireNamespace("DT", quietly = TRUE)) {
  install.packages("DT")
}

# Load the packages
library("dplyr")
library("ggplot2")

# We will consistently use package::function() to avoid confusion
# However, the dplyr package has to be loaded first to allow
# definitions of various operators such as %>%
```

## Data Preparation

Since our current analysis is dependent upon the differential gene expression
analysis from assignment 2, we will load the data from the previous assignment.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Load the differential gene expression analysis results from assignment 2

# Comparison between control and VTP without TGF-β1
deg_vtp_ctrl <- read.csv("./Data/dge_vtp_ctrl.csv", row.names = 1,
    header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# Comparison between control and VTP with TGF-β1
deg_vtp_tgf <- read.csv("./Data/dge_vtp_tgf.csv", row.names = 1,
    header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)
```

The data we just loaded are produced from the `topTags` function in the `edgeR`
package, which returns the log fold change, unadjusted p-value, and adjusted
p-value (FDR) for each gene.

# Gene Set Enrichment Analysis (GSEA)

We will perform GSEA on the differentially expressed genes between the control
and VTP treatment groups with and without the addition of TGF-β1 separately,
and compare the GSEA result with the over representation analysis (ORA) result
obtain previously from assignment 2.

## Effect of VTP treatment in the absence of TGF-β1

### Data Preparation

First, we aim to assess the effect of VTP treatment in the absence of TGF-β1 on
cardiac stromal cells by identifying its top enriched pathways. Since GSEA is a
non-threshold method, the use of a ranked list of genes without an arbitrarily
defined threshold allows the assessment of all genes in the list. The ranking of
the genes is integrates the sign of fold change change, and
the statistical significance of the change. [@subramanian2005gene] Since it has 
been suggested that 
GSEA is more consistent in identifying enriched gene sets and pathways, it is 
a preferable method to ORA for analyzing the two sets of differential gene 
expression results under two different conditions. [@abatangelo2009comparative]

We will generate a ranked list of genes as an input to the GSEA software. The
rank of each gene is calculated by the negative of log10 of the PValue by the 
sign of the logFC.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Add row names as a column for naming purpose
deg_vtp_ctrl$GeneName <- rownames(deg_vtp_ctrl)

# Constructing ranked gene list
deg_vtp_ctrl$rank <- (- log(deg_vtp_ctrl$PValue, base = 10)) * sign(deg_vtp_ctrl$logFC)
deg_vtp_ctrl_ranked <- dplyr::select(deg_vtp_ctrl, GeneName, rank) %>% 
    arrange(desc(rank)) %>%
    dplyr::select(GeneName, rank)

# Save the ranked gene list
write.table(deg_vtp_ctrl_ranked, file = "./Data/vtp_ctrl.rnk", sep = "\t", 
        col.name = TRUE, row.names = FALSE, quote = FALSE)
```


## Effect of VTP treatment in the presence of TGF-β1

### Data Preparation

In the previous analysis in assignment 2, the threshold over-representation
analysis (ORA) on the up- and down-regulated genes in the presence of TGF-β1
did not reveal any significant enriched gene sets due to the small number of genes
that were differentially expressed. However, GSEA is a non-threshold method, which
is not dependent on an arbitrary cutoff value. Therefore, it is expected to be
more sensitive than ORA and may in turn reveal some enriched pathways.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Add row names as a column for naming purpose
deg_vtp_tgf$GeneName <- rownames(deg_vtp_tgf)

# Constructing ranked gene list
deg_vtp_tgf$rank <- (- log(deg_vtp_tgf$PValue, base = 10)) * sign(deg_vtp_tgf$logFC)
deg_vtp_tgf_ranked <- dplyr::select(deg_vtp_tgf, GeneName, rank) %>% 
    arrange(desc(rank)) %>%
    dplyr::select(GeneName, rank)

# Save the ranked gene list
write.table(deg_vtp_tgf_ranked, file = "./Data/vtp_tgf.rnk", sep = "\t", 
        col.name = TRUE, row.names = FALSE, quote = FALSE)
```

## GSEA Analysis

The method

Since we are interested in understanding the effect of VTP treatment of the 
fibrotic phenotype/progression of the cardiac stromal cells, we are mainly
interested in understanding the biological processes that may be involved
in this process. Therefore, we used the most recent (March 2023) version of 
the `Human_GOBP_AllPathways_no_GO_iea_March_02_2023_symbol.gmt` gene set,
curated by the Bader Lab at the University of Toronto, available 
[here](http://download.baderlab.org/EM_Genesets/March_02_2023/Human/symbol/). 
[@reimand2019pathway] We specifically selected the gene set that does not 
contain annotations inferred from electronic annotations (IEA). This is to
ensure a high confidence inference of the biological processes from GSEA.





# Interpretation and Analysis


# Journal Link
The link to the concurrent journal entry can be found [here](https://github.com/bcb420-2023/Jielin_Yang/wiki/Gene-Set-Enrichment-and-Network-Analysis).

# References