---
title: "Gene Set Enrichement and Network Analysis"
subtitle: "BCB420 Assignment 3"
author: Jielin Yang
date: April 4th, 2023
output: 
  html_document:
    toc: true
    toc_depth: 3
    toc_float: true
    number_sections: true
    df_print: paged
    highlight: pygments
bibliography: Assignment3.bib
csl: "american-medical-association.csl"
---

<style type="text/css">
    body {
    max-width: 100%;
    overflow-x: auto;
    font-family: Helvetica;
    font-size: 12pt;
    }
    h1 {
    font-size: 20pt;
    font-weight: bold;
    }
    h2 {
        font-size: 18pt;
        font-weight: bold;
    }
    h3 {
        font-size: 16pt;
        font-weight: bold;
    }
    h4 {
        font-size: 14pt;
        font-weight: bold;
    }
    code {
        font-family: monospace;
        font-size: 11pt;
    }
</style>

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
knitr::opts_knit$set(root.dir = here::here())  # Must be the project directory
```

```{r message=FALSE, warning=FALSE, include=FALSE}
# Run A2 to generate the required data
rmarkdown::render("./A2_Jielin_Yang.Rmd")
```

# Introduction

## Background

Heart failure (HF) is a chronic cardiac disease with complex pathophysiology 
with profound implications in the patient's quality of life. The progressively
degenerating myocardium is a major hallmark of HF, which is accompanied by
cardiac fibrosis. [@garoffolo2022reduction] [@porter2009cardiac] [@tallquist2017redefining]
Therefore, the identification of profibrotic pathways is essential to the
understanding the pathophysiology of HF and the development of therapeutic
strategies to delay or even to prevent the profibrotic remodelling of the heart.

The Hippo pathway is a conserved signalling pathway that regulates cell
proliferation, cell survival, and cell migration. [@mia2019hippo] It has been
previously discovered that a central component of this pathway, the YAP/TAZ
complex, is dysregulated upon mechanimal stimulation of the cytoskeleton. 
[@elosegui2017force] [@panciera2017mechanobiology] [@brusatin2018biomaterials]
Interestingly, heart failure is usually accompanied by mechanical stress due to 
the malfunctioning of the pumping capability of the heart. Therefore, the
direct suppression of the YAP/TAZ complex may be a potential therapeutic
approach to suppress the proliferation of pro-fibrotic cells and to prevent
the profibrotic remodelling of the heart. [@garoffolo2022reduction] Recently, 
[Garoffolo et al.](https://www.ahajournals.org/doi/10.1161/CIRCRESAHA.121.319373) 
investigated the use of a small molecular inhibitor, verteporfin (VTP) in
cardiac stromal cells, an important precursor to cardiac fibroblasts, to 
suppress the YAP/TAZ complex with or without the presence of TGF-β1, a 
profibrotic cytokine under the maximal strain condition.

## Objectives

In the present analysis, we aim to use RNA-seq data to identify the changes in
the transcriptional profile of cardiac stromal cells after the use of VTP. As
shown in [Figure 1.](#fig1), the transcriptome was assessed by comparing the
addition of VTP with the control with or without the presence of the profibrotic
cytokine, TGF-β1. [@saadat2021pivotal] Upon the identification of differentially
expressed genes, we aim to perform gene set enrichment analysis (GSEA) and
network analysis to access the enriched pathways and using it to infer the
molecular mechanisms underlying the effects of VTP on cardiac stromal cells.

<a name="fig1"></a>
![](https://github.com/bcb420-2023/Jielin_Yang/raw/main/images/schematics.png)

**Figure 1.** Schematics of RNAseq experiment design. Samples were cultured under 
maximal mechanical strains. Six independent cell cultures were used for each 
treatment condition. cSt-Cs: cardiospheres-derived primitive cardiac stromal 
cells; VTP: verteporfin; TGF-β1: transforming growth factor beta-1. Figure 
adapted from [Assignment 1](https://github.com/bcb420-2023/Jielin_Yang/blob/main/Dataset%20selection%20and%20initial%20processing/Assignment_1_Data_cleaning_and_identifier_mapping.html) 
and created with BioRender.com.

## Dataset

The data used in the present analysis is retrieved from the GEO database under 
the accession number [GSE203358](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE203358). 
The dataset contains RNA-seq data from 24 samples, which are evenly split into 4 
groups:

- Control (maximal strain condition)
- Verteporfin (VTP) treatment (maximal strain condition)
- TGF-β1 treatment (maximal strain condition)
- Verteporfin (VTP) treatment with TGF-β1 (maximal strain condition)

## Previous Analysis

### Data Normalization

Upon filtering out genes with low counts, the dataset is 
composed of `14814` genes and `24` samples. The reference genome used is the 
human [GRCh38](https://useast.ensembl.org/Homo_sapiens/Info/Index?db=core).

The dataset was normalized using the Trimmed Mean of M-values (TMM) method 
implemented by the [edgeR](https://bioconductor.org/packages/release/bioc/html/edgeR.html) 
package. [@robinson2010edger] After applying normalization, the library size for 
each sample has been effectively adjusted such that the distribution of gene 
expressions are comparable across samples.

### Cluster Analysis

Samples were clustered based on transcriptional similarity. multidimensional
scaling analysis indicates distinct separation based on treatment conditions,
with the most highly distinguishable cluster of samples in the control group,
without profibrotic TGF-β1 treatment. Additional hierarchical clustering with
heatmap further supports that the control group has the most distinguishable
transcriptome, with few samples having similar expression patterns with the 
other groups. VTP treatment are found to alter the transcriptional profile 
to an high extent, but its effect is less prominent when combined with TGF-β1.
In addition, the samples show moderate to low degree of clustering based on the 
cell culture replicates, suggesting that the variation between samples are less
driven by the intrinsic biological variation between the cell cultures.

### Differential Expression Analysis and Preliminary ORA

Differential gene expression analysis was performed using the `edgeR` package. 
[@robinson2010edger] [Figure 2.](#fig2) shows the volcano plot of the differentially
expressed genes (DEGs) between the control and VTP treatment groups, with the
top enriched GO: Biological Process terms in the significantly downregulated 
genes. In this analysis, `r nrow(deg_vtp_ctrl)` genes are found to be differentially
expressed between the control and VTP treatment group, without the presence of
TGF-β1. Among these genes, `r nrow(deg_vtp_ctrl[deg_vtp_ctrl$logFC > 0,])` are
significantly upregulated and `r nrow(deg_vtp_ctrl[deg_vtp_ctrl$logFC < 0,])` are
significantly downregulated. Many of the downregulated genes are collagen genes,
which are a major component of the extracellular matrix (ECM) and are involved
in the profibrotic remodelling of the heart. [@garoffolo2022reduction] The top
enriched GO: Biological Process terms are also consistent with such findings, 
showing pathways in cell adhesion and ECM organization. However, when TGF-β1 is
added, the effect of VTP is much less pronounced, with only `r nrow(deg_vtp_tgf)`
genes being differentially expressed. 

<a name="fig2"></a>

```{r fig.height=4, fig.width=10, echo=FALSE, fig.align="center"}
# Replot the dot plot from assignment 2 with only the top 5 enriched GO terms
# in the significantly downregulated genes
gobp_downreg_5 <- gs_downreg_10[1:5,]
dot_plot_down <- ggplot2::ggplot(gobp_downreg_5) + 
                ggplot2::geom_point(ggplot2::aes(x = gene_ratio, 
                            y = reorder(stringr::str_wrap(Term, 20), gene_ratio),
                            color = FDR,
                            size = Fold.Enrichment)) +
                ggplot2::theme_bw() +
                ggplot2::theme(axis.text.x = ggplot2::element_text(size = 10),
                               axis.text.y = ggplot2::element_text(size = 10),
                               axis.title.x = ggplot2::element_text(size = 10),
                               axis.title.y = ggplot2::element_text(size = 10)) +
                ggplot2::scale_color_gradient(low = "red", high = "blue") +
                ggplot2::labs(x = "Gene Ratio", y = NULL,
                              color = "FDR", size = "Fold Enrichment")

# Replot the dot plot from assignment 2 with only the top 5 enriched GO terms
# in the significantly upregulated genes
gobp_upreg_5 <- gs_upreg_10[1:5,]
dot_plot_up <- ggplot2::ggplot(gobp_upreg_5) + 
                ggplot2::geom_point(ggplot2::aes(x = gene_ratio, 
                            y = reorder(stringr::str_wrap(Term, 20), gene_ratio),
                            color = FDR,
                            size = Fold.Enrichment)) +
                ggplot2::theme_bw() +
                ggplot2::theme(axis.text.x = ggplot2::element_text(size = 10),
                               axis.text.y = ggplot2::element_text(size = 10),
                               axis.title.x = ggplot2::element_text(size = 10),
                               axis.title.y = ggplot2::element_text(size = 10)) +
                ggplot2::scale_color_gradient(low = "red", high = "blue") +
                ggplot2::labs(x = "Gene Ratio", y = NULL,
                              color = "FDR", size = "Fold Enrichment")

# Replot the volcano plot from assignment 2
vol_plot <- ggplot2::ggplot(results_vtp_ctrl, aes(logFC, -log(FDR, 10))) +
    ggplot2::geom_point(aes(color = Expression), size = 2/5) +
    xlab(expression("log"[2]*"FC")) + 
    ylab(expression("-log"[10]*"FDR")) + 
    ggplot2::scale_color_manual(values = c("Up-regulated" = "firebrick3", 
                                           "Down-regulated" = "dodgerblue3", 
                                           "Not DE" = "gray50")) +
    ggplot2::guides(color = ggplot2::guide_legend(override.aes = list(size=1.5))) +
    ggrepel::geom_label_repel(data = genes_to_label, 
                        mapping = aes(logFC, -log(FDR, 10), label = rownames(genes_to_label)),
                        size = 2.5, color = "black", nudge_x = 0.1, nudge_y = 0.1)

# Draw the plots
cowplot::plot_grid(vol_plot, dot_plot_down, ncol = 2, labels = "AUTO")
```

**Figure 2.** Differentially expressed genes and enriched GO: Biological Process terms
in the significantly downregulated genes. (A) Volcano plot of the differentially
expressed genes between the control and VTP treatment groups in the absence of
TGF-β1. Important genes are labelled, with most collagen genes found to be 
downregulated. (B) Dot plot of the top 5 enriched GO: Biological Process terms
in the significantly downregulated genes.


# Preparations

## Package Dependencies

The current R Notebook assumes that it is being run on the BCB420-2023 base image. 
Additional packages required are noted below. For reproducibility we list all packages
we used in the analysis, including utility packages. The GSEA and Cytoscape softwares
with the EnrichmentMap pipeline are also required for the analysis, but they are
run manually outside of the R Notebook.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# ggplot2 package for creating plots
if (! requireNamespace("ggplot2", quietly = TRUE)) {
  install.packages("ggplot2")
}
# cowplot package for arranging plots
if (! requireNamespace("cowplot", quietly = TRUE)) {
  install.packages("cowplot")
}
# ggrepel package for plotting text with minimal overlap
if (! requireNamespace("ggrepel", quietly = TRUE)) {
  install.packages("ggrepel")
}
# stringr package for string manipulation
if (! requireNamespace("stringr", quietly = TRUE)) {
  install.packages("stringr")
}
# RCy3 package for interacting with Cytoscape
if (! requireNamespace("RCy3", quietly = TRUE)) {
  install.packages("RCy3")
}
# dplyr package for data manipulation
if (! requireNamespace("dplyr", quietly = TRUE)) {
  install.packages("dplyr")
}
# DT package for creating interactive tables
if (! requireNamespace("DT", quietly = TRUE)) {
  install.packages("DT")
}

# Load the packages
library("dplyr")
library("ggplot2")

# We will consistently use package::function() to avoid confusion
# However, the dplyr package has to be loaded first to allow
# definitions of various operators such as %>%
```

## Data Preparation

Since our current analysis is dependent upon the differential gene expression
analysis from assignment 2, we will load the data from the previous assignment.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Load the differential gene expression analysis results from assignment 2

# Comparison between control and VTP without TGF-β1
deg_vtp_ctrl <- read.csv("./Data/dge_vtp_ctrl.csv", row.names = 1,
    header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)

# Comparison between control and VTP with TGF-β1
deg_vtp_tgf <- read.csv("./Data/dge_vtp_tgf.csv", row.names = 1,
    header = TRUE, stringsAsFactors = FALSE, check.names = FALSE)
```

The data we just loaded are produced from the `topTags` function in the `edgeR`
package, which returns the log fold change, unadjusted p-value, and adjusted
p-value (FDR) for each gene.

# Gene Set Enrichment Analysis (GSEA)

We will perform GSEA on the differentially expressed genes between the control
and VTP treatment groups with and without the addition of TGF-β1 separately,
and compare the GSEA result with the over representation analysis (ORA) result
obtain previously from assignment 2.

## Effect of VTP treatment in the absence of TGF-β1

### Data Preparation

First, we aim to assess the effect of VTP treatment in the absence of TGF-β1 on
cardiac stromal cells by identifying its top enriched pathways. Since GSEA is a
non-threshold method, the use of a ranked list of genes without an arbitrarily
defined threshold allows the assessment of all genes in the list. The ranking of
the genes is integrates the sign of fold change change, and
the statistical significance of the change. [@subramanian2005gene] Since it has 
been suggested that 
GSEA is more consistent in identifying enriched gene sets and pathways, it is 
a preferable method to ORA for analyzing the two sets of differential gene 
expression results under two different conditions. [@abatangelo2009comparative]

We will generate a ranked list of genes as an input to the GSEA software. The
rank of each gene is calculated by the negative of log10 of the PValue by the 
sign of the logFC.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Add row names as a column for naming purpose
deg_vtp_ctrl$GeneName <- rownames(deg_vtp_ctrl)

# Constructing ranked gene list
deg_vtp_ctrl$rank <- (- log(deg_vtp_ctrl$PValue, base = 10)) * sign(deg_vtp_ctrl$logFC)
deg_vtp_ctrl_ranked <- dplyr::select(deg_vtp_ctrl, GeneName, rank) %>% 
    arrange(desc(rank)) %>%
    dplyr::select(GeneName, rank)

# Save the ranked gene list
write.table(deg_vtp_ctrl_ranked, file = "./Data/vtp_ctrl.rnk", sep = "\t", 
        col.name = TRUE, row.names = FALSE, quote = FALSE)
```

### GSEA Analysis

Gene set enrichment analysis (GSEA) was performed using the GSEA software (version
4.3.2) with the previously defined ranked gene list. We performed GSEA using the
predefined `GSEAPreranked` mode, which employs a default gene set permutation to
generate a null distribution of enrichment scores. [@subramanian2005gene] The
p-value of the enrichment is calculated using the Kolmogorov–Smirnov statistic,
with multiple hypothesis correction using FDR method. [@subramanian2005gene] [reimand2019pathway]
We performed GSEA analysis with the following parameters:

* Number of permutations: 1000
* Max size: 200
* Min size: 15

where `max size` and `min size` represent upper and lower bounds on the size of
the gene sets to be included in the analysis. Setting this threshold allowed
us to identify the pathways that are sufficiently specific for the interpretation 
of its biological function.

Since we are interested in understanding the effect of VTP treatment of the 
fibrotic phenotype/progression of the cardiac stromal cells, we are mainly
interested in understanding the biological processes that may be involved
in this process. Therefore, we used the most recent (March 2023) version of 
the `Human_GOBP_AllPathways_no_GO_iea_March_02_2023_symbol.gmt` gene set,
curated by the Bader Lab at the University of Toronto, available 
[here](http://download.baderlab.org/EM_Genesets/March_02_2023/Human/symbol/). 
[@reimand2019pathway] We specifically selected the gene set that does not 
contain annotations inferred from electronic annotations (IEA). This is to
ensure a high confidence inference of the biological processes from GSEA. The 
`gmt` file has been previously downloaded for [prior analysis](https://github.com/bcb420-2023/Jielin_Yang/wiki/GSEA). 
In this annotation file isc curates all GO: Biological Process terms as well as
pathways from multiple pathway databases, including Reactome, WikiPathways,
MSigDB. [@gillespie2022reactome] [@slenter2018wikipathways] [@liberzon2011molecular]
These annotation data should provide us an sufficient understanding of the enriched
biological responses to VTP treatment.

We use the following R code to run GSEA analysis.

```{r message = FALSE, tidy=TRUE, warning=FALSE, eval=FALSE}
# Configuring the GSEA software

# Path to the GSEA software
gsea_exe <- "/home/rstudio/projects/Jielin_Yang/GSEA_4.3.2/gsea-cli.sh"
# GSEA directory
gsea_dir <- "./Data/VTPControl/"
# Path to the gene set GMT file
gsea_gmt <- "./Data/Human_GOBP_AllPathways_no_GO_iea_March_02_2023_symbol.gmt"
# Path to the ranked gene list
gsea_rnk <- "./Data/vtp_ctrl.rnk"
# Permutations
perm <- 1000
# Max size
max_size <- 200
# Min size
min_size <- 15
# Analysis name
analysis_name <- "VTPControl"
```

```{r message = FALSE, tidy=TRUE, warning=FALSE, eval=FALSE}
# Running GSEA
cmd <- paste(gsea_exe, "GSEAPreranked", 
            "-gmx", gsea_gmt, 
            "-collapse false", 
            "-nperm", perm, 
            "-rnk", gsea_rnk,
            "-scoring_scheme weighted",
            "-rpt_label", analysis_name,
            "-plot_top_x 20 -rnd_seed 12345",
            "-set_max", max_size, 
            "-set_min", min_size, 
            "-zip_report false", 
            "-out", gsea_dir)
system(cmd)
```

### GSEA Results

Since we performed GSEA manually (details discussed in the [journal](https://github.com/bcb420-2023/Jielin_Yang/wiki/Gene-Set-Enrichment-and-Network-Analysis)), 
we will import the results from the `GSEA` software.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Import GSEA results for VTP vs Control
# Since R is having problem correctly parsing the tab in the tsv files, we have
# manually converted the results into a csv file to change the delimiters
# Detailed description documented in the journal
vtp_ctrl_gsea_upreg <- read.csv(
    "./Data/VTPControl/gsea_report_for_na_pos_1680318768172.csv", 
    header = TRUE)
vtp_ctrl_gsea_downreg <- read.csv(
    "./Data/VTPControl/gsea_report_for_na_neg_1680318768172.csv", 
    header = TRUE)
```


Our GSEA analysis identified ``r nrow(vtp_ctrl_gsea_upreg)`` out of `5256` 
specified gene sets associated with the upregulated genes in the VTP treatment 
condition. Of these, ``r sum(vtp_ctrl_gsea_upreg$FDR.q.val < 0.25)`` gene 
sets were significantly enriched (FDR < 0.25) with an average enrichment score
of ``r round(mean(vtp_ctrl_gsea_upreg$ES[vtp_ctrl_gsea_upreg$FDR.q.val < 0.25]), 2)``.
This signficance threshold was chosen in accordance with the default GSEA setting.
[@subramanian2005gene] 

[Table 1](#tab:1) shows the enriched gene sets with the corresponding enrichment
score and FDR q value. The top enriched processes are related to mitocondrial
functions, including mitochondrial translation as well as biological processes 
related to ribosomal functions. The enriched biological processes shows that 
the VTP treatment may be associated with an dystregualtion of metabolic processes
in the cells. 

<a name="tab1"></a>

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Filter and prepare the table for display
upreg_table <- vtp_ctrl_gsea_upreg %>% 
    dplyr::select(NAME, SIZE, ES, FDR.q.val) %>%
    dplyr::rename("Gene Set" = NAME, "Size" = SIZE, 
                  "Enrichment Score" = ES, "FDR q-value" = FDR.q.val) %>%
    # Extract the gene set name, round the enrichment score and FDR q-value
    dplyr::mutate(`Gene Set` = stringr::str_remove(`Gene Set`, "%.*$")) %>%
    # Convert the gene set name to sentence case
    dplyr::mutate(`Gene Set` = stringr::str_to_sentence(`Gene Set`)) %>%
    # Sort the table by FDR q-value
    dplyr::arrange(`FDR q-value`) %>%
    # Select significantly enriched gene sets
    dplyr::filter(`FDR q-value` < 0.25)

# Display the table
DT::datatable(upreg_table, rownames = FALSE, filter = "top", 
    options = list(ppageLength = 10, scrollX = TRUE)) %>%
    # Changing the font size of content
    DT::formatStyle(names(upreg_table), fontSize = "12px")
```

**Table 1.** Enriched gene sets in the upregulated genes under VTP treatment.



Furthermore, GSEA has also identified ``r nrow(vtp_ctrl_gsea_downreg)`` gene sets 
associated with the downregulated genes in the VTP. Of these, ``r sum(vtp_ctrl_gsea_downreg$FDR.q.val < 0.25)``
gene sets were significantly enriched (FDR < 0.25). The average enrichment score
for these downregulated gene sets is ``r round(mean(vtp_ctrl_gsea_downreg$ES[vtp_ctrl_gsea_downreg$FDR.q.val < 0.25]), 2)``.
There are more enriched gene sets identified in the downregulated genes compared
to the upregulated genes, which is expected since VTP is an inhibitor of the 
transcritpional activator complex YAP/TAZ. [@garoffolo2022reduction]
As shown in [Table 2](#tab:2), the top enriched gene sets are related to integrin
and collagen formaiton pathways. Since both are major components of the extracellular
matrix as well as cell adhesion, these enrichment results suggest that the VTP
has an effect on downregulating the processes involved in fibrosis formation.

<a name="tab2"></a>

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Filter and prepare the table for display
downreg_table <- vtp_ctrl_gsea_downreg %>% 
    dplyr::select(NAME, SIZE, ES, FDR.q.val) %>%
    dplyr::rename("Gene Set" = NAME, "Size" = SIZE, 
                  "Enrichment Score" = ES, "FDR q-value" = FDR.q.val) %>%
    dplyr::mutate(`Gene Set` = stringr::str_remove(`Gene Set`, "%.*$")) %>%
    dplyr::mutate(`Gene Set` = stringr::str_to_sentence(`Gene Set`)) %>%
    dplyr::arrange(`FDR q-value`) %>%
    dplyr::filter(`FDR q-value` < 0.25)

# Display the table
DT::datatable(downreg_table, rownames = FALSE, filter = "top", 
    options = list(ppageLength = 10, scrollX = TRUE)) %>%
    DT::formatStyle(names(downreg_table), fontSize = "12px")
```

**Table 2.** Enriched gene sets in the downregulated genes under VTP treatment.

### Comparative analysis of GSEA and ORA

[Figure 3](#fig:3) illustrates the top enriched gene sets identified by GSEA and ORA,
respectively for the up- and down-regualted genes. Qualitatively, the results
from the two analysis are similar in terms of the top enriched gene sets.
Both types of analysis identified dysregulation of mitochondrial functions in 
the top enriched gene sets. However, ORA reveal additional terms relating to
cellular responses to stress and external stimuli, which are either not identified
or has a comparatively lower enrichment score and significance value in GSEA. 
This may be due to the fact that when we performed ORA, we specificially limited
the genes to have a fold change of at least 2 
([Yang J. BCB420 Assignment 2](https://github.com/bcb420-2023/Jielin_Yang/blob/main/A2_Jielin_Yang.html)). 
Putting such a limit may have
resulted in the increased proportion of genes relating to cellular stress response,
allowing ORA to identify these terms as top enriched gene sets. On the other hand,
both analyses revealed a consistant enrichment of gene sets related to collagen
formation and ECM remodeling in the downregulated genes. Since VTP is an inhibitory
drug that targets a specific activator complex, the resulting inhibitory effect
is expected to be more prominent than the stimulatory effect. Therefore, having
both types of analyses showing a consistant enrichment in downregulated genes
not only indicates that the VTP has an effect on downregulating the processes
involved in fibrosis formation, but also provides a more robust evidence that
reveals the consistency of the two types of analysis when using dataset showing
a clear and strong differential expression.


<a name="fig3"></a>

```{r message = FALSE, tidy=TRUE, warning=FALSE, fig.align="center", fig.width=10, fig.height=7}
# Top 5 enriched gene sets of upregulated genes, GSEA
gsea_up <- ggplot2::ggplot(upreg_table[1:5,], aes(x = `Enrichment Score`, 
    y = stringr::str_wrap(`Gene Set`, 20))) +   
    geom_bar(stat = "identity",) +
    labs(x = "Enrichment Score", y = NULL) +
    theme_minimal() +
    theme(axis.line.x = element_line(),
        axis.text.x = element_text(size = 10),
        axis.text.y.left = element_text(size = 10),
        axis.text.y = element_text(hjust = 0))

# Top 5 enriched gene sets of downregulated genes, GSEA
downreg_table$`Gene Set` <- gsub("_", " ", downreg_table$`Gene Set`)  # Replace underscore with space
gsea_down <- ggplot2::ggplot(downreg_table[1:5,], aes(x = -`Enrichment Score`, 
    y = stringr::str_wrap(`Gene Set`, 20))) +   
    geom_bar(stat = "identity",) +
    labs(x = "Enrichment Score (absolute value)", y = NULL) +
    theme_minimal() +
    theme(axis.line.x = element_line(),
        axis.text.x = element_text(size = 10),
        axis.text.y.left = element_text(size = 10),
        axis.text.y = element_text(hjust = 0))

# We have previously generated the plot for the ORA results in Assignment 2,
# and we reuse it for clear comparison
cowplot::plot_grid(dot_plot_up, gsea_up, dot_plot_down, gsea_down, 
    ncol = 2, nrow = 2, labels = "AUTO", rel_heights = c(1, 1))
```

**Figure 3.** Top 5 enriched gene sets identified by GSEA and ORA for the up- 
(A, B) and down-regulated (C, D) genes. Results from previous ORA was shown in
A and C, with colour of the dots indicates significance level, and the size indicates
the fold enrichment. B, D illustrate the top 5 enriched gene sets identified by
GSEA, with the enrichment score on the x-axis. All GSEA results are coloured
the same as all gene sets has a FDR q-value of 0 (up to the maximum signficant figure).

However, such qualitative comparason between the results from GSEA and ORA
are neither straitforward nor conclusive. In this case, we have to compare top
enriched gene sets to access the consistency between the most significant results.
However, in both types of analysis, both the order of the significance value as 
well as the effect sizes (enrichment score and fold enrichment. respectively)
also critically effect our interpretation of the results. By qualitatively
comparing them, it is unlikely for human eyes to identify the hierarchical
structure returned by the two types of analysis, especially when it comes to
gene sets beyond the first few top enrichments.


## Effect of VTP treatment under TGF-β1

Since the authors of the original publication also investigated whether the effect
of VTP is preseved even under the profibrotic environment, we also performed
GSEA based on the differential gene expression analysis results comparing the
VTP-treated cardiac stromal cells and control under TGF-β1 treatment. 

### Data Preparation

While there are not many differntially expressed genes in this dataset,
we calcualted the rank of each gene similarly to the previous analysis.

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Add row names as a column for naming purpose
deg_vtp_tgf$GeneName <- rownames(deg_vtp_tgf)

# Constructing ranked gene list
deg_vtp_tgf$rank <- (- log(deg_vtp_tgf$PValue, base = 10)) * sign(deg_vtp_tgf$logFC)
deg_vtp_tgf_ranked <- dplyr::select(deg_vtp_tgf, GeneName, rank) %>% 
    arrange(desc(rank)) %>%
    dplyr::select(GeneName, rank)

# Save the ranked gene list
write.table(deg_vtp_tgf_ranked, file = "./Data/vtp_tgf.rnk", sep = "\t", 
        col.name = TRUE, row.names = FALSE, quote = FALSE)
```

### GSEA Analysis

Similarly, we performed GSEA analysis using the same parameters as the previous
analysis, except that this time we use the `vtp_tgf.rnk` gene list for the analysis.
Only a few parameters needs to be changed. 

```{r message = FALSE, tidy=TRUE, warning=FALSE, eval=FALSE}
# Change parameters

# Rank file
gsea_rnk <- "./Data/vtp_tgf.rnk"
# GSEA directory
gsea_dir <- "./Data/VTPTGF/"
# Analysis name
analysis_name <- "VTPTGF"

# Run GSEA
cmd <- paste(gsea_exe, "GSEAPreranked", 
            "-gmx", gsea_gmt, 
            "-collapse false", 
            "-nperm", perm, 
            "-rnk", gsea_rnk,
            "-scoring_scheme weighted",
            "-rpt_label", analysis_name,
            "-plot_top_x 20 -rnd_seed 12345",
            "-set_max", max_size, 
            "-set_min", min_size, 
            "-zip_report false", 
            "-out", gsea_dir)
system(cmd)
```

### GSEA Result 

```{r message = FALSE, tidy=TRUE, warning=FALSE}
# Import GSEA results for up and down regulated genes under TGF-β1 treatment
# Similarly, we have manually converted the GSEA results to a csv file for proper import
gsea_vtp_tgf_up <- read.csv("./Data/VTPTGFPreranked/gsea_report_for_na_pos_1680319898865.csv",
    header = TRUE)
gsea_vtp_tgf_down <- read.csv("./Data/VTPTGFPreranked/gsea_report_for_na_neg_1680319898865.csv", 
    header = TRUE)
```

[Figure 4](#fig4) summarizes the top enriched gene sets in the up- and down-
regulated genes under TGF-β1 treatment. In this case, the top enriched gene sets
for both up- and down-regulated genes are consistent with the GSEA result without
the presence of TGF-β1. In summary, ``r nrow(gsea_vtp_tgf_up)`` gene sets are
enriched in the up-regulated genes, with ``r sum(gsea_vtp_tgf_up$FDR.q.val < 0.25)`` 
of them being significant. For down-regualted genes, ``r sum(gsea_vtp_tgf_down$FDR.q.val < 0.25)``
out of ``r nrow(gsea_vtp_tgf_down)`` enriched gene sets are found to be signifiant.

<a name="fig4"></a>

```{r message = FALSE, tidy=TRUE, warning=FALSE, fig.align="center", fig.width=10, fig.height=4}
# Filter the top 5 enriched gene sets, retain only fields of interest
gsea_vtp_tgf_up <- gsea_vtp_tgf_up[1:5, c("NAME", "ES", "FDR.q.val")]
gsea_vtp_tgf_down <- gsea_vtp_tgf_down[1:5, c("NAME", "ES", "FDR.q.val")]

# Extract the gene set names and convert to sentence case
gsea_vtp_tgf_up$NAME <- stringr::str_remove(gsea_vtp_tgf_up$NAME, "%.*$") %>%
    stringr::str_to_sentence()
gsea_vtp_tgf_down$NAME <- stringr::str_remove(gsea_vtp_tgf_down$NAME, "%.*$") %>%
    stringr::str_to_sentence()

# Plot the top 5 enriched gene sets
gsea_vtp_tgf_up <- ggplot2::ggplot(gsea_vtp_tgf_up, aes(x = ES, y = NAME)) +   
    geom_bar(stat = "identity") +
    geom_text(aes(x = 0.01, y = NAME, label = NAME), size = 5, hjust = 0) +
    labs(x = "Enrichment Score", y = "Term") +
    theme_minimal() +
    theme(axis.line.x = element_line(),
        axis.text.x = element_text(size = 11),
        axis.text.y = element_blank())

gsea_vtp_tgf_down <- ggplot2::ggplot(gsea_vtp_tgf_down, aes(x = -ES, y = NAME)) +
    geom_bar(stat = "identity") +
    geom_text(aes(x = 0.01, y = NAME, label = NAME), size = 5, hjust = 0) +
    labs(x = "Enrichment Score (absolute value)", y = "Term") +
    theme_minimal() +
    theme(axis.line.x = element_line(),
        axis.text.x = element_text(size = 11),
        axis.text.y = element_blank())

# Combine the two plots
cowplot::plot_grid(gsea_vtp_tgf_up, gsea_vtp_tgf_down, ncol = 2, labels = "AUTO")
```

**Figure 4.** Top 5 enriched gene sets under TGF-β1 for A) up- and B) down-
regulated genes with VTP treatment. The length of the bars represent the
absolute value of the enrichment score, and the gene set names are shown
on the bars.


Further examination on the result reveals that the enriched terms computed
by GSEA are consistent between the two datasets (with and without TGF-β1). Since
GSEA calculates enrichment score based on the ranking of the genes, it follows
that the ranking in both cases are similar. [@reimand2019pathway] [@subramanian2005gene]
Although in the presence of TGF-β1, there are only a few differentially expressed
genes, the difference between the GSEA result with the previous ORA analysis,
which returned no enriched gene set, reveals that GSEA is a more sensitive measure
to detect functional enrichment of the genes. Thus, GSEA is not limited to the
lack of differentially expressed genes, and it can reveals the global change in
the pattern of gene expression once VTP treatment is applied, despite to a smaller
effect. Therefore, GSEA varifies that under TGF-β1 treatment, VTP treatment still
induces an overall antifibrotic change on the transcriptome of the cardiac 
stromal cells.

Since the GSEA analysis result reveals a consistent pattern of functional enrichment
in both datasets (with and without TGF-β1), we will focus our analysis on the
enrichment results without TGF-β1 treatment. This would allow us to concentrate
on potential pathways that are significantly dysregualted by VTP treatment, and 
having a clearly defined set of differentially expressed genes would further allow
us to perform a robust network analysis.

# Network Analysis

## Network Construction

As detailed in the [journal](https://github.com/bcb420-2023/Jielin_Yang/wiki/Gene-Set-Enrichment-and-Network-Analysis#network-analysis),
we have tried to run EnrichmentMap from within R, but we have encountered several
issues. The `RCy3` package was unable to establish a connection with Cytoscape
from within Docker, and if we run R directly on the host machine, syntax errors
exist that current version of the Cytoscape/EnrichmentMap plugin may use a slightly
different syntax than the version in which the [tutorial](https://baderlab.github.io/Cytoscape_workflows/EnrichmentMapPipeline/Protocol2_createEM.html#28_Create_an_Enrichment_map) was written. Therefore,
we have decided to run Cytoscape manually from the GUI.










<a name="fig5"></a>
![](https://github.com/bcb420-2023/Jielin_Yang/raw/main/images/VTP_ctrl_network_no_format.svg)



# Interpretation and Analysis


# Journal Link
The link to the concurrent journal entry can be found [here](https://github.com/bcb420-2023/Jielin_Yang/wiki/Gene-Set-Enrichment-and-Network-Analysis).

# References