---
title: "Data cleaning and identifier mapping"
subtitle: "Assignment 1 for BCB420 Winter 2023"
author: Jielin Yang
date: Feb. 14th, 2023
output: 
  html_document:
    toc: true
    toc_depth: 3
    toc_float: true
    number_sections: true
    df_print: paged
    highlight: pygments
bibliography: Assignment1.bib
csl: /home/rstudio/projects/Jielin_Yang/american-medical-association.csl
---

<style type="text/css">
    body {
    max-width: 100%;
    overflow-x: auto;
    font-family: Helvetica;
    font-size: 12pt;
    }
    h1 {
    font-size: 20pt;
    font-weight: bold;
    }
    h2 {
        font-size: 18pt;
        font-weight: bold;
    }
    h3 {
        font-size: 16pt;
        font-weight: bold;
    }
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```
    


# Introduction

## Background

Heart failure (HF) is one of the most common and life-threatening syndrome 
associated with progressive morbidity and mortality that has implicated over 65 
million patients worldwide in 2017. [@savarese2022global] It is characterized by 
the reduced capacity of the heart to pump sufficient blood to meet the metabolic
demands of the body. The pathogenesis of HF is complex and involves multiple 
factors, but one of the earliest hallmark of the disease is maladaptive
remodelling of the myocardium. [@garoffolo2022reduction] During remodelling, the
damaged cardiomyocytes (which can be resulted from ventricular overload, altered
metabolic composition, etc.) are replaced by fibrotic scar tissues.
 [@porter2009cardiac] [@tallquist2017redefining] The altererd extracellular 
matrix (ECM) increased myocardial stiffness and thus further
contributes to the progression of HF. [@perestrelo2021multiscale] Therefore, the
identification of the key genes and pathways that are involved in profibrotic 
remodeling is critical for the development of novel therapeutic teachniques.

Earlier studies have established a connection between both static and dynamic 
cytoskeletal tension between cells as a result of cellular adhesion to ECM with 
the transactivation of the Hippo signaling pathway. [@brusatin2018biomaterials] 
In particular, it was found that an increased nuclear entry of the YAP/TAZ 
complex, which is a potent transcriptional activator of proliferative genes, is 
a result of the inhibition of the repressive kinease regulators and deformation 
of the nuclear lamina under mechanical strains. [@panciera2017mechanobiology] 
[@elosegui2017force]
Recently, this finding has been extended to myofibroblasts, which are the major 
contributor of cardiac fibrosis derived from the quiescence cardiac stromal cells
. [@francisco2020blockade] Hence, 
[Garoffolo et al.](https://doi.org/10.1161/CIRCRESAHA.121.319373) aims to 
elicidate the connection between the abnormal straining forces in cardiac 
myocardium and the YAP-dependent transcriptional activation in profibrotic 
cells. [@garoffolo2022reduction] In addition, they tested whether direct 
pharmaceutical inhibition of the YAP/TAZ complex with verteporfin (VTP), a known
drug that interferes the binding of YAP to its target TEAD, can attenuate the 
progression of cardiac fibrosis with and without the presence of the potent 
profibrotic cytokine TGF-β1. [@giraud2020verteporfin] [@piersma2015signaling]

## Objectives
The research study aimed to establish the connection between the response of 
human myocardial stromal cells (cSt-Cs) to straining forces and the activation 
of the profibrotic pathway downsteam of YAP/TAZ. In addition, they tested 
whether the inhibition of YAP/TAZ translocation to the nucleus. 
[@garoffolo2022reduction]

Here, we focused on a particular part of the larger objective, where we 
incorporated the RNA sequencing data to differentially compare the molecular 
signitures of the cSt-Cs under known mechanical strains that activate 
profibrotic pathways.
We hope to identify whether the direct inactivation of the YAP/TAZ complex with 
VTP under *in vitro* conditions is able to attenuate the progression of cardiac 
fibrosis, with and without the presence of TGF-β. [@garoffolo2022reduction]

## Data
As documented in the previous [journal](https://github.com/bcb420-2023/Jielin_Yang/wiki/Gene-Expression-Dataset-Selection) 
and [R Notebook](https://github.com/bcb420-2023/Jielin_Yang/blob/main/Data%20set%20selection%20and%20initial%20processing/Gene_Expression_Dataset_Selection.Rmd),
we have performed a search for gene expression datasets that are related to the 
topic of our research project using both the GEOmetadb and the NCBI GEO database.
Upon reviewing the results, we have selected the dataset with the accession number [GSE203358](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE203358) for 
further analysis.
This dataset is part of a study that has been published in Circulation Research.
The original publication can be found 
[here](https://doi.org/10.1161/CIRCRESAHA.121.319373).

## Study Summary
### Publication Title
Reduction of Cardiac Fibrosis by Interference With YAP-Dependent Transactivation

### General Experimental Design
To access whether direct inhibition of the YAP/TAZ complex contributes to
anti-fibrotic activities under chronic mechanical strain and profibrotic 
signalling, RNA sequencing was performed in the following way. 
Cardiospheres-derived primitive cardiac stromal cells (cSt-Cs) were derived from
cardiospheres retrieved from human myocardial explant tissues. cSt-Cs were then 
treated with ±verteporfin (VTP) and ±TGF-β1 under *in vitro* culture conditions 
under maximum strain exerted by glass slides.
Total RNA was extracted from each independent cell culture 72 hours after
phamalogical treatments using the Trizol method. RNA library was made following 
enrichment of the polyadenylated mRNAs.

```{r echo = FALSE, fig.align = 'center', out.width = "80%", fig.cap = "Schematics of RNAseq experiment design. Samples were cultured under maximal mechanical strains. Six independent cell cultures were used for each treatment condition. cSt-Cs:  cardiospheres-derived primitive cardiac stromal cells; VTP: verteporfin; TGF-β1: transforming growth factor beta-1. Figure created with BioRender.com"}
knitr::include_graphics("schematics.png")
```

# Preparations

The current R Notebook assumes that it is being run on the BCB420-2023 base 
image. Additional packages required are noted below.

```{r, message = FALSE, tidy=TRUE}
# BiocManager for installing a set of Bioconductor packages
if (! requireNamespace("BiocManager", quietly = TRUE)) {
  install.packages("BiocManager")
}

# GEOquery package for querying the GEO database
if (! requireNamespace("GEOquery", quietly = TRUE)) {
  BiocManager::install("GEOquery")
}

# edgeR package for manupulating and analyzing RNAseq data
if (! requireNamespace("edgeR", quietly = TRUE)) {
  BiocManager::install("edgeR")
}

# Biobase package for Bioconduction base functions
if (! requireNamespace("Biobase", quietly = TRUE)) {
  BiocManager::install("Biobase")
}

# KableExtra package for creating tables and formatting
if (! requireNamespace("kableExtra", quietly = TRUE)) {
  install.packages("kableExtra")
}

# dplyr package for data manipulation
if (! requireNamespace("dplyr", quietly = TRUE)) {
  install.packages("dplyr")
}

# Load the dplyr package
library("dplyr")
```


# Data Download and Initial Characterization

## Dataset Description

In the previous section, we have discussed the study design of the RNA sequencing.
The RNAseq data generated in this study constitutes the second part of the
larger study, where the authors tested direct pharmaceutical inhibition of the
YAP/TAZ complex with verteporfin (VTP), after revealing that mechanical straining
of the cardiac stromal cells (cSt-Cs) activates the YAP/TAZ complex. 
[@garoffolo2022reduction] To better understand the dataset itself, here we 
retrieve the GEO description of the dataset.

```{r, message = FALSE, tidy=TRUE}
# Accession number
geoAcc <- "GSE203358"

# Retrieve the metadata given the GSE series accession number
# Set GESMatrix to FALSE as it is specific to microarray data
gse <- GEOquery::getGEO(geoAcc, GSEMatrix = FALSE)

# Further metadata can be retrieved regarding the platform used
gpl <- names(GEOquery::GPLList(gse))[1]
gplInfo <- GEOquery::Meta(GEOquery::getGEO(gpl))
```

Summary of the dataset metadata:

- **GEO Accession Number:** GSE203358
- **Title:** `r GEOquery::Meta(gse)$title`
- **Platform:** `r gplInfo$title`
- **Submission date:** `r gplInfo$submission_date`
- **Last update date:** `r GEOquery::Meta(gse)$last_update_date`
- **Organism:** `r paste(gplInfo$organism, paste0("(taxid: ", gplInfo$taxid, ")"), collapse = " ")`
- **Number of samples:** `r length(GEOquery::GSMList(gse))`
- **Link to BioProject:** [`r substring(GEOquery::Meta(gse)$relation, first = 13)`](`r substring(GEOquery::Meta(gse)$relation, first = 13)`)
- **Overall study design (by the [authors](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE203358)):** `r GEOquery::Meta(gse)$overall_design`


## Data Download
The description of the dataset shows that there are a total of 24 samples, with
6 replicates for each of the 4 treatment conditions. We will download the raw
data from the GEO database and store it in a local directory.

```{r, message = FALSE, tidy=TRUE}
# A `./Data` directory has been created to store the data.
# We will download the raw data using the GEOquery package.

# Check if the data file exists, and if not, download and decompress it
if (! file.exists("./Data/GSE203358_Count_Matrix.txt")) {
  sfiles = GEOquery::getGEOSuppFiles(geoAcc, makeDirectory = FALSE, 
                                     baseDir = "./Data")
  fnames = rownames(sfiles)
  GEOquery::gunzip(fnames[1])
}
```

Only one data file exists for this study. The raw counts for all 24 samples are
located in the file: `r list.files(path = "./Data", pattern = "\\.txt")`

## Preliminary Data Checking
The raw counts are stored in a tab-delimited text file. We will import the data
into R using the `read.table()` function.

```{r, message = FALSE, tidy=TRUE}
# Get the unzipped file name
file <- list.files(path = "./Data", pattern = "\\.txt")

# Import the raw counts into R
rawCounts <- read.table(paste0("./Data/", file), sep = "\t", 
                        header = TRUE, check.names = FALSE)

colnames(rawCounts)
```

There are 25 columns in the dataset, with the first column as gene ID, and 
the rest with 1 column per sample

```{r, message = FALSE, tidy=TRUE}
# Rename the rows with gene ID
rownames(rawCounts) <- rawCounts$Geneid

# Visualize the first 5 genes
head(rawCounts)
```

The dataset does not contain the original Ensembl gene IDs, whereas the HUGO
gene symbols are used.


# Data QC and Processing

## Sample Matrix

First we will retrieve the smaple design matrix, and retrieve the replicate
names under each of the four conditions

```{r, message = FALSE, tidy=TRUE}
# Obtain the sample names and treatment conditions using the column names from
# the dataset
samples <- data.frame(lapply(colnames(rawCounts)[2:25], 
                             function(x){unlist(strsplit(x, split = "\\."))}))

# rename the dataframe
colnames(samples) <- colnames(rawCounts)[2:25]
rownames(samples) <- c("culture_rep","treatment")

# Perform a matrix transposition, so that individual samples are listed in rows
# Construct a data frame of the sample design matrix
samples <- data.frame(t(samples))

# Order the samples by their treatment conditions
samples <- samples[order(samples$treatment), ]

samples
```

## Mapped Genes

### Gene duplication

We will first assess whether there are any duplicated genes in the dataset.

```{r, message = FALSE, tidy=TRUE}
# How many genes are mapped in the data set?
dim(rawCounts)
```

There are a total of `r dim(rawCounts)[1]` rows in the dataset, indicating the
number of genes mapped. We will now deduplicate them and see if there is any 
duplicated the genes.

```{r, message = FALSE, tidy=TRUE}
# Number of unique genes mapped
length(unique(rawCounts$Geneid))

# Whether there are any dupilcated genes
!(length(unique(rawCounts$Geneid)) == dim(rawCounts)[1])    # No duplication
```

### Gene mapping

Previously, we did not observe any duplicated genes in the dataset based on the gene names.
According to the method section the publication, total RNA was extracted from the cells
and subjected to library preparation using the poly-A enrichment method. Therefore, it
is expected that the genes selected for sequencing and analysis are protein-coding genes.
We will now check the highly expressed genes in the dataset, and see if these genes 
are consistent with our expectation. Since we only have gene names rather than
Ensembl gene ID, we can directly search for and understand potential functions of 
these highly expressed genes. While we are not performing differential gene 
expression analysis, understanding the most highly enriched genes would allow us
to check whether the general gene expression profile matches with the experimental
condition, i.e, cells cultured under profibrotic mechanical strains.

```{r, message = FALSE, tidy=TRUE}
# Calculate the total counts for each gene and sort them in descending order
sumGeneCounts <- rowSums(rawCounts[, 2:25])
names(sumGeneCounts) <- rawCounts$Geneid
sumGeneCounts <- sort(sumGeneCounts, decreasing = TRUE)

# Genes with expression values > 1, and show the top 10
knitr::kable(sumGeneCounts[sumGeneCounts > 1][1:10], format = "html", 
             col.names = "Frequency") %>% 
            kableExtra::kable_styling("striped", position = "center", 
                                      full_width = F)
```

The most highly expressed gene is `FN1`, which encodes Fibronectin, which has been
found to be highly enriched in the extracellular matrix (ECM) of the heart tissue
following myocardial injury. [@wang2013fibronectin] Additionally, we have seen
`COL1A1`, `COL1A2` genes, which encode different chains of type 1 collagen that
are a common marker for fibrotic scars. [@hua2020multi] [@umbarkar2021mechanisms]
The gene expression profile is consistent with the expectation, as the highly
expressed genes (raw counts) across all biological replicates and test conditions
are found to be associated with fibrosis and ECM remodelling.

### Summary Statistics

We will then calculate the summary statistics for the raw counts for each 
of the 24 samples.

```{r, message = FALSE, tidy=TRUE}
# Calculate the summary statistics for the raw counts
summaryStats <- data.frame(apply(rawCounts[, 2:25], 2, summary))

# Visualize with a table
summaryStats
```

Since the samples are sequenced multiplexed in the same lane, under a randomized 
assumption, these samples should have similar sequencing depth. There are slight
variations in terms of the global distribution of the raw counts. We will check if
these variations are present between the different treatment conditions.

```{r, message = FALSE, tidy=TRUE}
# Calculate the summary statistics for the raw counts for each treatment condition

# Control
summaryCtrl <- summary(rowMeans(rawCounts[, grep("CTRL", colnames(rawCounts))]))

# TGFb only
summaryTGFb <- summary(rowMeans(rawCounts[, grep("TGFb$", colnames(rawCounts))]))

# VTP only
summaryVTP <- summary(rowMeans(rawCounts[, grep("VTP$", colnames(rawCounts))]))

# VTP + TGFb
summaryVTP_TGFb <- summary(rowMeans(rawCounts[, grep("TGFb_VTP", colnames(rawCounts))]))

# Combine the data frames and visualize
sumStats <- rbind(summaryCtrl, summaryTGFb, summaryVTP, summaryVTP_TGFb)
rownames(sumStats) <- c("CTRL", "TGFb", "VTP", "VTP_TGFb")
knitr::kable(sumStats, format = "html", col.names = colnames(sumStats)) %>% 
            kableExtra::kable_styling("striped", position = "center")
```

The mean values of the raw counts are similar across the different treatment. This 
suggests that the previously observed variations in the raw counts are mainly 
due to the biological replicates.


### Data Filtering
Since the presence of lowly expressed genes would not contribute to the downstream
analysis, which would even decrese the statistical power and sensitivity of detecting 
differentially expressed genes, we will filter out the genes with low gene counts.
[@sha2015effect]

According to the edgeR protocol, we will filter out the genes with no more than 
1 read per million in 6 of the samples, because for each treatment condition we
have 4 biological replicates. [@anders2013count]

```{r, message = FALSE, tidy=TRUE}
# Translating the raw counts into counts per million (CPM)
cpm <- edgeR::cpm(rawCounts[, 2:25])
# and rename the gene names
rownames(cpm) <- rawCounts$Geneid

# Filter out the genes with no more than 1 read per million in 6 of the samples
cpm_filtered <- cpm[rowSums(cpm > 1) >= 6, ]

# Visualize the summary statistics for the filtered dataset
dim(cpm_filtered)
```

It is observed that `r (dim(rawCounts)[1] - dim(cpm_filtered)[1])/dim(rawCounts)[1] * 100`% of the genes were filtered out.
Only `r dim(cpm_filtered)[1]` (`r dim(cpm_filtered)[1]/dim(rawCounts)[1] * 100`%) genes were retained for downstream analysis.


# Interpretation and Analysis
- What are the control and test conditions of the dataset?


- Why is the dataset of interest to you?


- Were there expression values that were not unique for specific genes? How did you handle these?


- Were there expression values that could not be mapped to current HUGO symbols?


- How many outliers were removed?


- How did you handle replicates?


- What is the final coverage of your dataset?










TODO:
1. For R code chucks, set warnings to false


# Journal Link
The link to the concurrent journal entry can be found [here](https://github.com/bcb420-2023/Jielin_Yang/wiki/Dataset-Cleaning-and-Identifier-Mapping).

# References